### PR TITLE
Update public API docstrings to align with the style guidelines

### PR DESCRIPTION
**Pull request by edmorley**
This is a port/sync of the style changes made to the SDK docs in the docs repository as part of the review cycle in:
https://github.com/salesforcedocs/salesforce-functions/pull/183

GUS-W-12477898.

### CODE REVIEW

File: salesforce_functions/data_api/__init__.py [hunk 1]
```diff
@@ -167,7 +168,7 @@ async def commit_unit_of_work(
         self, unit_of_work: UnitOfWork
     ) -> dict[ReferenceId, str]:
         """
-        Commit a `UnitOfWork`, executing all operations registered with it.
+        Commit a `UnitOfWork`, which executes all operations registered with it.
```
Comment (dzuelke) [inline salesforce_functions/data_api/__init__.py:171]: IMO this is now ambiguous (does the commit execute all operations or does the `UnitOfWork`?)

File: salesforce_functions/data_api/unit_of_work.py [hunk 2]
```diff
@@ -14,7 +14,13 @@ class UnitOfWork:
     """
     Represents a `UnitOfWork`.
 
-    After registering all operations, it can be submitted via the `commit_unit_of_work` method of `DataAPI`.
+    A `UnitOfWork` encapsulates a set of one or more Salesforce operations that must be
+    performed as a single atomic operation. Single atomic operations reduce the number of
+    requests back to the org, and are more efficient when working with larger data volumes.
+
+    First register the create, update, or delete operations that make up the `UnitOfWork`
```
Comment (dzuelke) [inline salesforce_functions/data_api/unit_of_work.py:21]: "First, …"

**Review by dzuelke — APPROVED** [review]

File: salesforce_functions/data_api/unit_of_work.py (hunk 2 above)
Comment (edmorley) [reply salesforce_functions/data_api/unit_of_work.py]: f2758d1261ecc9e0610ff756a7a4ca7c14f00d8b

File: salesforce_functions/data_api/__init__.py (hunk 1 above)
Comment (edmorley) [reply salesforce_functions/data_api/__init__.py:171]: This is the wording change that they requested (and has already been made to the generated docs), and whilst I agree the "which" is ambiguous, I think it doesn't detract from the ability to understand what `commit_unit_of_work()` does, so I don't think it's worth another round trip between two repos to change further :-)

**Pull request merged**